src/TwigComponent/src/Twig/ComponentExtension.php

```diff
@@ -84,6 +89,43 @@ public function finishEmbeddedComponentRender(): void
         $this->container->get(ComponentRenderer::class)->finishEmbeddedComponentRender();
     }
 
+    public function cva(array &$context, string $recipeName, array $recipe): void
+    {
+        $key = '__recipe__'.$recipeName;
+
+        $context[$key] = CVA::init($recipe);
```

Comment:
Do you think it is possible to use the Twig filename to combine it to the context key?

I'm afraid of conflicts between _too-much generic name_ recipe name across multiple components.

**EDIT:** or this is what `needs_context` and `$context` already do?

Reply:
Interesting, I've learned something today :d

Reply:
Actually not a specific file, but every component live inside his own context